pkg/helpers/crd.go

```diff
@@ -102,3 +131,85 @@ func (s SecretSortList) Less(i, j int) bool {
 
 	return s.Items[i].CreationTimestamp.Before(&s.Items[j].CreationTimestamp)
 }
+
+// Checks if the slice of strings contains a specific string
+func SliceContainsString(list []string, s string) bool {
+	for _, v := range list {
+		if v == s {
+			return true
+		}
+	}
+	return false
+}
+
+// handleDeletion will handle the finalizers if the object was deleted.
+// It will return true, if the deletiontimestamp is set. If the object was
+// removed the reconcile can be returned.
+// The resource to be removed can be nil, in case there's no direct resource,
+// like for a cluster without Vault.
+func HandleDeletion(instance metav1.Object, resource Remover, finalizerName string, client client.Client) (bool, error) {
+	if instance.GetDeletionTimestamp() == nil {
+		return false, nil
+	}
+
+	annotationValue, exists := instance.GetAnnotations()[DeleteProtectionAnnotation]
+
+	// this is really weird: https://play.golang.org/p/NJkbjN7fCMa
```

Comment:
It's because you shadow `protected` in the scope of the `if`. See https://play.golang.org/p/_e2pkyyzamD